src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore.hasOwnProperty('hidden') === true
+      && herbivore.hidden !== true) {
+      herbivore.health -= 50;
+
+      if (herbivore.health === 0) {
```

Comment:
handle negative values as well
```suggestion
      if (herbivore.health <= 0) {
```

Reply:
done